spec/bump/commands/preview_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "spec_helper"
+
+describe Bump::CLI::Commands::Preview do
+  it 'calls the server and display the link on success' do
+    stub_and_return('previews/post_success.http')
+
+    expect do
+      new_command.call(file: __FILE__, format: 'yaml')
+    end.to output(/Preview created/).to_stdout
+
+    expect(WebMock).to have_requested(:post,'https://bump.sh/api/v1/previews')
+  end
+
+  it 'displays the definition errors in case of unprocessable entity' do
+    stub_and_return('previews/post_unprocessable_entity.http')
+
+    expect do
+      begin
+        new_command.call(file: __FILE__, format: 'yaml')
+      rescue SystemExit; end
+    end.to output(/Definition is not valid/).to_stdout
+
+    expect(WebMock).to have_requested(:post,'https://bump.sh/api/v1/previews')
+  end
+
+  it 'displays the error message in case of system error' do
+    stub_and_return('previews/post_system_error.http')
+
+    expect do
+      begin
+        new_command.call(file: __FILE__, format: 'yaml')
+      rescue SystemExit; end
+    end.to output(/Something bad happened./).to_stderr
+
+    expect(WebMock).to have_requested(:post,'https://bump.sh/api/v1/previews')
+  end
+
+  private
+
+  def stub_and_return(path)
```

Comment:
This method names confused me. I was wondering what's stubbed and looking for the `return` but actually it's just a stub for an API call. Naming it `stub_bump_api_create_preview` could be better?

Reply:
:+1: